### ISSUE TITLE
API to manipulate certain types of values such as lists and maps 

### ISSUE THREAD

**Issue by aghassemi** [vanadium/issues#742]
An interesting use-case that was brought up by Kishore was ability to append to a value which is an array without getting the whole thing, manipulating it and doing a put. 

These sort of direct manipulations for certain types like arrays, maps and sets can be very useful.
For instance for array we can have:  `GetAtIndex, Append, RemoveAtIndex, ChangeAtIndex.`

After talking to @kenashcraft , we already have a plan for this which is @jakline adding `Insert` and `Update` support in the query language so one can reach into values as much as the query language allows. So technically we already have `GetAtIndex` functionality and support Insert and `Update` would allow `ChangeAtIndex` and Append.

This issue is to discuss whether we should create a higher-level API on top of the query for manipulating common types. For instance something like:

```
arrayRow := table.ArrayRow(key) // we can either fail here if value it not an array or fail on the first operation.
arrayRow.Append(value) 
arrayRow.Change(index, newValue)
etc...
```

@asadovsky

**Comment by asadovsky** [comment]
Yeah, I see this as something that would belong in the query language. E.g. note that the mongodb query language, couchdb n1ql (http://blog.couchbase.com/introducing-n1ql-sql-gene-in-nosql), etc. support all sorts of things along these lines. It's par-for-the-course with stores where values are structured.

I think @jakline has plans along these lines, but I'm not sure whether the specific ops you called out would be supported.
